internal/transactions/streaming.go

```diff
@@ -0,0 +1,236 @@
+
+	return nil
+}
+
+// Stop stops the transaction streamer
+func (ts *TransactionStreamer) Stop() {
+	close(ts.stopChan)
+}
+
+// pollMempool polls the mempool for new transactions
+func (ts *TransactionStreamer) pollMempool(ctx context.Context) {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// Get mempool content
+			var result map[string]map[string]map[string]*types.Transaction
+			err := ts.rpcClient.Call(&result, "txpool_content")
+			if err != nil {
+				fmt.Printf("Error getting mempool content: %v\n", err)
+				continue
+			}
+
+			// Process queued transactions
+			for _, account := range result["queued"] {
+				for _, tx := range account {
+					ts.processNewTransaction(ctx, tx)
+				}
+			}
+
+			// Process pending transactions
+			for _, account := range result["pending"] {
+				for _, tx := range account {
+					ts.processNewTransaction(ctx, tx)
+				}
+			}
+		case <-ctx.Done():
+			return
+		case <-ts.stopChan:
+			return
+		}
+	}
+}
+
+// processNewTransaction processes a new transaction from the mempool
+func (ts *TransactionStreamer) processNewTransaction(ctx context.Context, tx *types.Transaction) {
+	// Check if we already have this transaction
+	ts.mu.RLock()
+	_, exists := ts.txStore[tx.Hash().Hex()]
+	ts.mu.RUnlock()
+
+	if exists {
+		return
+	}
+
+	// Get full transaction details
+	_, _, err := ts.client.TransactionByHash(ctx, tx.Hash())
+	if err != nil {
+		fmt.Printf("Error getting transaction details: %v\n", err)
+		return
+	}
+
+	// Get sender address
+	sender, err := types.Sender(types.LatestSignerForChainID(tx.ChainId()), tx)
+	if err != nil {
+		fmt.Printf("Error getting sender address: %v\n", err)
+		return
+	}
+
+	// Create new stored transaction
+	storedTx := &StoredTransaction{
+		Hash: tx.Hash().Hex(),
+		Metadata: TransactionMetadata{
+			Status:       StatusQueued,
+			TimeReceived: time.Now().Unix(),
+			Type:         getTransactionType(tx),
+			Nonce:        tx.Nonce(),
+			From:         sender.Hex(),
+			To:           tx.To().Hex(),
+			Timestamp:    time.Now().Unix(),
+		},
+	}
+
+	// Set gas price based on transaction type
+	switch tx.Type() {
+	case types.DynamicFeeTxType:
+		storedTx.Metadata.MaxFeePerGas = tx.GasFeeCap()
+		storedTx.Metadata.MaxPriorityFee = tx.GasTipCap()
+	case types.BlobTxType:
+		storedTx.Metadata.MaxFeePerGas = tx.GasFeeCap()
+		storedTx.Metadata.MaxPriorityFee = tx.GasTipCap()
+		storedTx.Metadata.MaxFeePerBlobGas = tx.BlobGasFeeCap()
+	default:
+		storedTx.Metadata.GasPrice = tx.GasPrice()
+	}
+
+	// Store transaction
+	ts.mu.Lock()
+	ts.txStore[storedTx.Hash] = storedTx
+	ts.mu.Unlock()
+}
+
+// trackTransactionStatus tracks the status of transactions
+func (ts *TransactionStreamer) trackTransactionStatus(ctx context.Context) {
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			ts.mu.RLock()
+			for _, tx := range ts.txStore {
```

Comment:
Highlighted in previous comment: This should be fetching entries from the redis queue `queued`.